### PR TITLE
Sitemap REST API: bypass proxy for HLS video playlist

### PR DESCRIPTION
**Pull request by lolodomo**
Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

File: bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/SitemapResource.java [hunk 1]
```diff
@@ -471,16 +472,20 @@ private WidgetDTO createWidgetBean(String sitemapName, Widget widget, boolean dr
...
-            if (uri.getPort() < 0 || uri.getPort() == 80) {
-                bean.url = uri.getScheme() + "://" + uri.getHost() + "/proxy?sitemap=" + sitemapName
-                        + ".sitemap&widgetId=" + wId;
+            if (videoWidget.getEncoding() != null && videoWidget.getEncoding().toLowerCase().contains("hls")) {
+                bean.url = videoWidget.getUrl();
             } else {
-                bean.url = uri.getScheme() + "://" + uri.getHost() + ":" + uri.getPort() + "/proxy?sitemap="
-                        + sitemapName + ".sitemap&widgetId=" + wId;
+                String wId = itemUIRegistry.getWidgetId(widget);
+                if (uri.getPort() < 0 || uri.getPort() == 80) {
```
Comment (htreu) [inline bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/SitemapResource.java]: matter of taste: can this be refactored to use a `StringBuilder`? like:

```
StringBuilder sb = new StringBuilder();
sb.append(uri.getScheme()).append("://").append(uri.getHost());
if (uri.getPort() < 0 || uri.getPort() == 80) {
    sb.append(":").append(uri.getPort());
}
sb.append("/proxy?sitemap=").append(sitemapName).append(".sitemap&widgetId=").append(wId);
bean.url = sb.toString();
```

Comment (lolodomo) [reply bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/SitemapResource.java]: Yes and a method could ne created as the same code is used twice, once for image and once for video.

**Comment by lolodomo** [comment]
Changed as requested.

File: bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/SitemapResource.java [hunk 2]
```diff
@@ -510,6 +501,17 @@ private WidgetDTO createWidgetBean(String sitemapName, Widget widget, boolean dr
         return bean;
     }
 
+    private String buildProxyUrl(String sitemapName, Widget widget, URI uri) {
+        String wId = itemUIRegistry.getWidgetId(widget);
+        StringBuilder sb = new StringBuilder();
+        sb.append(uri.getScheme()).append("://").append(uri.getHost());
+        if (uri.getPort() >= 0 && uri.getPort() != 80) {
```
Comment (maggu2810) [inline bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/SitemapResource.java]: You moved this method only to another method, but is it really correct?
If the scheme is http the default port is 80, so all is fine.
If the scheme is https the default port should be 443 and so `uri.getPort()` should be checked to be 443.
Okay, this only affect the strange situation if someone runs https on port 80, so we should perhaps never run into.
But shouldn't it be fixed?

Comment (lolodomo) [reply bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/SitemapResource.java]: We could set the port in URL everytime ?

Comment (maggu2810) [reply bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/SitemapResource.java]: I agree. If it is the default one, it does not matter..

**Comment by lolodomo** [comment]
Change done.

**Pull request merged**